trl/trainer/ppov2_trainer.py

```diff
@@ -0,0 +1,732 @@
+import os
+import time
+from dataclasses import dataclass
+from typing import Dict, List, Literal, Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from accelerate import Accelerator
+from accelerate.state import AcceleratorState
+from accelerate.utils import broadcast
+from datasets import Dataset
+from torch.utils.data import DataLoader
+from transformers import (
+    DataCollatorWithPadding,
+    GenerationConfig,
+    PreTrainedModel,
+    PreTrainedTokenizer,
+    Trainer,
+    TrainerCallback,
+    TrainerControl,
+    TrainerState,
+    TrainingArguments,
+)
+from transformers.integrations import get_reporting_integration_callbacks
+from transformers.trainer_callback import CallbackHandler, DefaultFlowCallback
+
+
+INVALID_LOGPROB = 1.0
+
+
+"""
+python -i trl/trainer/ppov2_trainer.py \
+    --learning_rate 3e-6 \
+    --output_dir models/minimal/ppo \
+    --per_device_train_batch_size 1 \
+    --gradient_accumulation_steps 64 \
+    --lr_scheduler_type linear \
+"""
+
+
+@dataclass
+class PPOConfig(TrainingArguments):
+    # common args
+    exp_name: str = os.path.basename(__file__)[: -len(".py")]
+    """the name of this experiment"""
+    run_name: Optional[str] = None
+    """a unique name of this run"""
+    deepspeed2: bool = False
+    """Whether to use deepspeed to train the model"""
+    deepspeed3: bool = False
+    """Whether to use deepspeed to train the model"""
+
+    # various batch sizes
+    world_size: Optional[int] = None
```

Comment:
Actually yes, the `world_size` is set by `accelerate` later. I put it here for experiment tracking purposes. For example `world_size` helps calculate the `args.batch_size` (which is the effective / global batch size across all processes).

Reply:
I see, but why do we only seem to need this for the RL trainers and not the other ones like `SFTTrainer`? In general, I'd like to avoid exposing this distributed stuff to the user if we can because it might not be clear if they should set the value manually or let `accelerate` handle it for them